File: Ejercicios_POO3.1_Java/src/Ejercicio5_Tareas/Tarea.java
package Ejercicio5_Tareas;

public class Tarea {
    private String nombre;
    private boolean completada;

    public Tarea(String nombre) {
        this.nombre = nombre;
        this.completada = false;
    }

    public String getNombre() {
        return nombre;
    }

    public boolean isCompletada() {
        return completada;
    }

    public void marcarComoCompletada() {
        this.completada = true;
    }

    @Override
    public String toString() {
        return "Tarea{" +
                "nombre='" + nombre + '\'' +
                ", completada=" + completada +
                '}';
    }
}

File: Ejercicios_POO3.1_Kotlin/src/Ejercicio5_Tareas/Tarea.kt
package Ejercicio5_Tareas

data class Tarea(val nombre: String) {
    var completada: Boolean = false
        private set

    fun marcarComoCompletada() {
        completada = true
    }

    override fun toString(): String {
        return "Tarea(nombre='$nombre', completada=$completada)"
    }
}